domain/src/test/java/org/oppia/android/domain/classify/rules/fractioninput/FractionInputIsEquivalentToAndInSimplestFormRuleClassifierProviderTest.kt

```diff
@@ -0,0 +1,283 @@
+
+    assertThat(matches).isFalse()
+  }
+
+  // Even if creator does not input simplest form, learner's answer must be in simplest form
+  @Test
+  fun testEquivalentAndSimplest_fraction1Over2Answer_fraction4Over8Input_matches() {
+    val inputs = mapOf("f" to FRACTION_VALUE_TEST_4_OVER_8)
+    val answer = FRACTION_VALUE_TEST_1_OVER_2
+
+    val matches =
+      inputIsEquivalentToAndInSimplestFormRuleClassifier.matches(answer = answer, inputs = inputs)
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testEquivalentAndSimplest_fraction4Over8Answer_fraction4Over8Input_doesNotMatch() {
+    val inputs = mapOf("f" to FRACTION_VALUE_TEST_4_OVER_8)
+    val answer = FRACTION_VALUE_TEST_4_OVER_8
+
+    val matches =
+      inputIsEquivalentToAndInSimplestFormRuleClassifier.matches(answer = answer, inputs = inputs)
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testEquivalentAndSimplest_fraction1Over2Answer_negativeFraction1Over2Input_doesNotMatch() {
+    val inputs = mapOf("f" to NEGATIVE_FRACTION_VALUE_TEST_1_OVER_2)
+    val answer = FRACTION_VALUE_TEST_1_OVER_2
+
+    val matches =
+      inputIsEquivalentToAndInSimplestFormRuleClassifier.matches(answer = answer, inputs = inputs)
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testEquivalentAndSimplest_mixedNum16And1Over2Answer_mixedNum6And1Over2Input_matches() {
+    val inputs = mapOf("f" to MIXED_NUMBER_VALUE_TEST_16_AND_1_OVER_2)
+    val answer = MIXED_NUMBER_VALUE_TEST_16_AND_1_OVER_2
+
+    val matches =
+      inputIsEquivalentToAndInSimplestFormRuleClassifier.matches(answer = answer, inputs = inputs)
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testEquivalentAndSimplest_mixedNum16And1Over2Answer_mixedNum7And1Over7Input_doesNotMatch() {
+    val inputs = mapOf("f" to MIXED_NUMBER_VALUE_TEST_7_AND_1_OVER_7)
+    val answer = MIXED_NUMBER_VALUE_TEST_16_AND_1_OVER_2
+
+    val matches =
+      inputIsEquivalentToAndInSimplestFormRuleClassifier.matches(answer = answer, inputs = inputs)
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testEquivalentAndSimplest_mixedNum16And2Over4Answer_mixedNum6And1Over2Input_doesNotMatch() {
+    val inputs = mapOf("f" to MIXED_NUMBER_VALUE_TEST_16_AND_1_OVER_2)
+    val answer = MIXED_NUMBER_VALUE_TEST_16_AND_2_OVER_4
+
+    val matches =
+      inputIsEquivalentToAndInSimplestFormRuleClassifier.matches(answer = answer, inputs = inputs)
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testEquivalentAndSimplest_mixedNum16And1Over2Ans_negMixedNum16And1Over2Input_doesNotMatch() {
+    val inputs = mapOf("f" to NEGATIVE_MIXED_NUMBER_VALUE_TEST_16_AND_1_OVER_2)
+    val answer = MIXED_NUMBER_VALUE_TEST_16_AND_1_OVER_2
+
+    val matches =
+      inputIsEquivalentToAndInSimplestFormRuleClassifier.matches(answer = answer, inputs = inputs)
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testEquivalentAndSimplest_missingInputF_throwsException() {
```

Comment:
Suggest also adding tests for the wrong type of input/wrong type of answer (see other classifier tests for a reference).

Reply:
Done